pkg/sql/opt/norm/factory.go

```diff
@@ -340,5 +340,5 @@ func (f *Factory) ConstructConstVal(d tree.Datum, t *types.T) opt.ScalarExpr {
 		}
 		return memo.FalseSingleton
 	}
-	return f.ConstructConst(d)
+	return f.ConstructConst(d, t)
```

Comment:
Should we just kill this method altogether, and then add a couple rules for converting `(Const Null)` to `(Null)`, `(Const True)` to `(True)`, etc?